### PR TITLE
JE-46109 add onBeforeInstallAddon/onAfterInstallAddon

### PR DESCRIPTION
**Pull request by vlobzakov**
JE-46109 add onBeforeInstallAddon/onAfterInstallAddon events. Requires to add missed descriptions of some parameters

### CODE REVIEW

File: docs/creating-manifest/events.md [hunk 1]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
 ```
 @@!
 
+### onBeforeInstallAddon  
+The event is executed before add-on installation  
+
+Event Placeholders:  
+
+-   `${event.params.}`:  
+    - `settings` - ***What is it***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: values of http://docs.cloudscripting.com/creating-manifest/visual-settings/

File: docs/creating-manifest/events.md [hunk 2]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+The event is executed before add-on installation  
+
+Event Placeholders:  
+
+-   `${event.params.}`:  
+    - `settings` - ***What is it***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 3]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+Event Placeholders:  
+
+-   `${event.params.}`:  
+    - `settings` - ***What is it***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 4]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+-   `${event.params.}`:  
+    - `settings` - ***What is it***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 5]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it***  
+    - `id` - add-on id  
+    - `lang` - ***What is it***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: user's language code (https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter

File: docs/creating-manifest/events.md [hunk 6]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+    - `lang` - ***What is it***  
+    - `region` - region, where an environment will be installed  
+    - `targetAppid`  - application identifier  
+-   `${event.response.}`: parameters are absent  
+
+### onAfterInstallAddon  
+The event is executed after add-on installation  
+  
+-   `${event.params.}`:  
+    - `settings` - ***What is it?***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: values of http://docs.cloudscripting.com/creating-manifest/visual-settings/

File: docs/creating-manifest/events.md [hunk 7]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+
+### onAfterInstallAddon  
+The event is executed after add-on installation  
+  
+-   `${event.params.}`:  
+    - `settings` - ***What is it?***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 8]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+The event is executed after add-on installation  
+  
+-   `${event.params.}`:  
+    - `settings` - ***What is it?***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it?***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 9]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+-   `${event.params.}`:  
+    - `settings` - ***What is it?***  
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it?***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it?***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: remove

File: docs/creating-manifest/events.md [hunk 10]
```diff
@@ -1757,7 +1757,109 @@ onAfterRemoveSSL:
...
+    - `envGroups` - environment groups array  
+    - `displayName` - display name for an environment  
+    - `skipNodeEmails` - an ability to skip sending emails about creating nodes. Emails are related only to nodes where implemented reset password functionality  
+    - `source` - ***What is it?***  
+    - `nodeGroup` - the defined node layer  
+    - `exportedFromEnv` - ***What is it?***  
+    - `envName` - short domain name (without hosting provider URL)  
+    - `actionKey` - ***What is it?***  
+    - `id` - add-on id  
+    - `lang` - ***What is it?***
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: user's language code (https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter

**Review by SlavaKatiukha — CHANGES_REQUESTED** [review]

File: docs/creating-manifest/events.md (hunk 1 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: - `settings` - custom form with predefined user input elements

Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 2 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 3 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 4 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 5 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 6 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 7 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 8 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 9 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 10 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done.

**Review by vlobzakov — COMMENTED** [review]
Requested changes done. Please review the line: https://github.com/vlobzakov/docs/edit/vlobzakov-JE-46109-onBeforeInstallAddon-onAfterInstallAddon/docs/creating-manifest/events.md#L1784 is it correct?

File: docs/creating-manifest/events.md [hunk 11]
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
...
+    - `id` - add-on id  
+    - `lang` - [user's language code](https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter  
+    - `region` - region, where an environment will be installed  
+    - `targetAppid`  - application identifier  
+-   `${event.response.}`:  
+    - `uniqueName` - installed add-on unique name  
+    - `appid` - unique environment *appid* at the Jelastic Platform  
+    - `successText`- message that appears once action is successfully performed  
+  
+Subscription:
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: change to 'Events Subscription Example'. Just 'Subscription' is confusing a little bit.

File: docs/creating-manifest/events.md [hunk 12]
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
...
+    - `successText`- message that appears once action is successfully performed  
+  
+Subscription:  
+  
+@@@
+```yaml
+type: update  
+name: Test Add-on Events Subscription  
+  
+onBeforeInstallAddon [nodeGroup:cp, id:test123]:
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: replace *test123* with *yourAddonId* (all occurrences)

**Review by SlavaKatiukha — CHANGES_REQUESTED** [review]

File: docs/creating-manifest/events.md (hunk 11 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done

File: docs/creating-manifest/events.md (hunk 12 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: I replaced addon id 'test123' with 'myaddon'. 
And I propose to add a 'unique' word in front of add-on id for better understanding that add-ons' ids cannot coincide

**Review by vlobzakov — COMMENTED** [review]
please check the updates

File: docs/creating-manifest/events.md [hunk 13]
```diff
@@ -1757,7 +1757,93 @@ onAfterRemoveSSL:
...
+
+### onAfterInstallAddon  
+The event is executed after add-on installation  
+  
+-   `${event.params.}`:  
+    - `settings` - custom form with predefined user input elements   
+    - `id` - unique add-on id  
+    - `lang` - [user's language code](https://en.wikipedia.org/wiki/Language_code) which was used during the installation. Optional parameter  
+    - `region` - region, where an environment will be installed  
+    - `targetAppid`  - application identifier
```
Comment (SlavaKatiukha) [inline docs/creating-manifest/events.md]: change to "unique environment *appid* at the Jelastic Platform"

**Review by SlavaKatiukha — CHANGES_REQUESTED** [review]

File: docs/creating-manifest/events.md (hunk 13 above)
Comment (vlobzakov) [reply docs/creating-manifest/events.md]: done:
https://github.com/vlobzakov/docs/pull/19/commits/794d0b8d68d7c3a24eddcc5433116766cc222e61
https://github.com/vlobzakov/docs/pull/19/commits/1ffa9f07646e16b7a20380bc3f4a29c34db5067b

**Review by SlavaKatiukha — APPROVED** [review]

**Pull request merged**